src/main/java/org/broadinstitute/consent/http/service/UserService.java

```diff
@@ -456,6 +460,15 @@ private void addExistingLibraryCards(User user) {
     userDAO.updateUser(user.getDisplayName(), user.getUserId(), user.getInstitutionId());
   }
 
+  private void populateLibraryCardsOnUsers(List<User> users) {
+    for (User user: users) {
+      List<LibraryCard> cards = libraryCardDAO.findLibraryCardsByUserId(user.getUserId());
+      if (Objects.nonNull(cards) && !cards.isEmpty()) {
+        user.setLibraryCards(cards);
+      }
+    }
```

Comment:
This is a flag for me when I see a DAO call in a loop. Instead of 1 database call for users with LCs, we're now making N+1 database calls. Ideally, we should be doing the population in `userDAO.getUsersFromInstitutionWithCards()` and `userDAO.findUsersWithLCsAndInstitution()`. Those methods should be fully populating the DAA associations as part of the library card - their names suggest as much. They are a bit thorny, but I think it's worth it to refactor them into text block queries and ditch the query field construction pattern for both of them.